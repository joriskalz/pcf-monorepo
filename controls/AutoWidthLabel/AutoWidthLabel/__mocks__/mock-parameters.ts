/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
