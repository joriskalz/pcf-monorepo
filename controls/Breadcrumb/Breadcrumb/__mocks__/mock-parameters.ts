/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
