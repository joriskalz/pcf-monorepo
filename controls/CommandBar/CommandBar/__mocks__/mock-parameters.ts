/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
