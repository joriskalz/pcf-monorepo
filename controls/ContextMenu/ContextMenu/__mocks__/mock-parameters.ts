/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
