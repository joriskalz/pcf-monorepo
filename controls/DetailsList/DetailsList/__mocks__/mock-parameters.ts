/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
