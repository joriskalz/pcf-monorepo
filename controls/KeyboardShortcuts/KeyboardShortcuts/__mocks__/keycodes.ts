export class KeyCodes {
  static BACKSPACE = 8;
  static TAB = 9;
  static ENTER = 13;
  static SHIFT = 16;
  static CTRL = 17;
  static ALT = 18;
  static PAUSE = 19;
  static CAPS_LOCK = 20;
  static ESCAPE = 27;
  static SPACE = 32;
  static PAGE_UP = 33;
  static PAGE_DOWN = 34;
  static END = 35;
  static HOME = 36;
  static LEFT_ARROW = 37;
  static UP_ARROW = 38;
  static RIGHT_ARROW = 39;
  static DOWN_ARROW = 40;
  static INSERT = 45;
  static DELETE = 46;
  static KEY_0 = 48;
  static KEY_1 = 49;
  static KEY_2 = 50;
  static KEY_3 = 51;
  static KEY_4 = 52;
  static KEY_5 = 53;
  static KEY_6 = 54;
  static KEY_7 = 55;
  static KEY_8 = 56;
  static KEY_9 = 57;
  static KEY_A = 65;
  static KEY_B = 66;
  static KEY_C = 67;
  static KEY_D = 68;
  static KEY_E = 69;
  static KEY_F = 70;
  static KEY_G = 71;
  static KEY_H = 72;
  static KEY_I = 73;
  static KEY_J = 74;
  static KEY_K = 75;
  static KEY_L = 76;
  static KEY_M = 77;
  static KEY_N = 78;
  static KEY_O = 79;
  static KEY_P = 80;
  static KEY_Q = 81;
  static KEY_R = 82;
  static KEY_S = 83;
  static KEY_T = 84;
  static KEY_U = 85;
  static KEY_V = 86;
  static KEY_W = 87;
  static KEY_X = 88;
  static KEY_Y = 89;
  static KEY_Z = 90;
  static LEFT_META = 91;
  static RIGHT_META = 92;
  static SELECT = 93;
  static NUMPAD_0 = 96;
  static NUMPAD_1 = 97;
  static NUMPAD_2 = 98;
  static NUMPAD_3 = 99;
  static NUMPAD_4 = 100;
  static NUMPAD_5 = 101;
  static NUMPAD_6 = 102;
  static NUMPAD_7 = 103;
  static NUMPAD_8 = 104;
  static NUMPAD_9 = 105;
  static MULTIPLY = 106;
  static ADD = 107;
  static SUBTRACT = 109;
  static DECIMAL = 110;
  static DIVIDE = 111;
  static F1 = 112;
  static F2 = 113;
  static F3 = 114;
  static F4 = 115;
  static F5 = 116;
  static F6 = 117;
  static F7 = 118;
  static F8 = 119;
  static F9 = 120;
  static F10 = 121;
  static F11 = 122;
  static F12 = 123;
  static NUM_LOCK = 144;
  static SCROLL_LOCK = 145;
  static SEMICOLON = 186;
  static EQUALS = 187;
  static COMMA = 188;
  static DASH = 189;
  static PERIOD = 190;
  static FORWARD_SLASH = 191;
  static GRAVE_ACCENT = 192;
  static OPEN_BRACKET = 219;
  static BACK_SLASH = 220;
  static CLOSE_BRACKET = 221;
  static SINGLE_QUOTE = 222;
}
