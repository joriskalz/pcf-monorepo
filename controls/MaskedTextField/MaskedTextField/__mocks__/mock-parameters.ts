/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Value: new MockStringProperty(),
        Prefix: new MockStringProperty(),
        Suffix: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        ErrorMessage: new MockStringProperty(),
        Mask: new MockStringProperty(),
        MaskFormat: new MockStringProperty(),
        DelayOutput: new MockTwoOptionsProperty(),
    };
}
