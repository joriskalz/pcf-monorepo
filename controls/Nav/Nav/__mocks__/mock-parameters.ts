/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        CollapseByDefault: new MockTwoOptionsProperty(),
        InputEvent: new MockStringProperty(),
        SelectedKey: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
