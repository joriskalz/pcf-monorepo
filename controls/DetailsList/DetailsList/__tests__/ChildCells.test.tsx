import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { IGridColumn } from '../Component.types';
import { GridCell, GridCellProps } from '../GridCell';
import { MockEntityRecord } from '../__mocks__/mock-datasets';

const item = new MockEntityRecord('1', {
    col1: 'Row 1 Col 1',
    col2: 'Row 1 Col 2',
    col3: 'Row 1 Col 3',
    col4: 'Row 1 Col 4',
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
} as any);

const defaultProps: GridCellProps = {
    item: item,
    index: 1,
    column: {
        key: 'col1',
        fieldName: 'col1',
        name: 'Column 1',
        minWidth: 200,
        maxWidth: 400,
        calculatedWidth: 200,
        childColumns: [
            {
                key: 'col2',
                fieldName: 'col2',
                name: 'Column 2',
                showAsSubTextOf: 'col1',
                subTextRow: 0,
            } as IGridColumn,
        ],
    } as IGridColumn,

    onCellAction: () => {
        //noop
    },
};

describe('GridCell', () => {
    it('renders child cells in root cell content', () => {
        const mockProps: GridCellProps = {
            ...defaultProps,
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            item: item,
        };
        const component = renderer.create(<GridCell {...mockProps} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders child cells in root cell content followed by additional child cells', () => {
        const mockProps: GridCellProps = {
            ...defaultProps,
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            item: item,
            column: {
                ...defaultProps.column,
                childColumns: [
                    // Sub Row 0
                    {
                        key: 'col2',
                        fieldName: 'col2',
                        name: 'Column 2',
                        showAsSubTextOf: 'col1',
                        subTextRow: 0,
                    } as IGridColumn,
                    // Sub Row 1
                    {
                        key: 'col3',
                        fieldName: 'col3',
                        name: 'Column 3',
                        showAsSubTextOf: 'col1',
                        subTextRow: 1,
                    } as IGridColumn,
                    {
                        key: 'col4',
                        fieldName: 'col4',
                        name: 'Column 4',
                        showAsSubTextOf: 'col1',
                        subTextRow: 1,
                    } as IGridColumn,
                ],
            } as IGridColumn,
        };
        const component = renderer.create(<GridCell {...mockProps} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
