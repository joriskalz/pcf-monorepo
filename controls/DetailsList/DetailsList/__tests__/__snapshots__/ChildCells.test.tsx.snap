// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridCell renders child cells in root cell content 1`] = `
<div
  className=

      {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
      }
>
  <span
    className=

        {
          font-weight: 400;
          max-width: 400px;
          overflow: hidden;
          padding-right: 4px;
          text-overflow: ellipsis;
          white-space: nowrap;
        }
  >
    Row 1 Col 1
  </span>
  <span
    className=

        {
          font-weight: 400;
        }
  >
    Row 1 Col 2
  </span>
</div>
`;

exports[`GridCell renders child cells in root cell content followed by additional child cells 1`] = `
<div
  className=

      {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
      }
>
  <span
    className=

        {
          font-weight: 400;
          max-width: 400px;
          overflow: hidden;
          padding-right: 4px;
          text-overflow: ellipsis;
          white-space: nowrap;
        }
  >
    Row 1 Col 1
  </span>
  <span
    className=

        {
          font-weight: 400;
        }
  >
    Row 1 Col 2
  </span>
  <span
    className="sub-text-row-wrapper-break"
  />
  <span
    className=

        {
          font-weight: 400;
          padding-right: 4px;
        }
  >
    Row 1 Col 3
  </span>
  <span
    className=

        {
          font-weight: 400;
        }
  >
    Row 1 Col 4
  </span>
</div>
`;
